robottelo/ui/contentviews.py

```diff
@@ -450,6 +462,45 @@ def add_remove_errata_to_filter(self, cv_name, filter_name,
         else:
             self.click(locators['contentviews.remove_errata'])
 
+    def edit_erratum_date_range_filter(
+            self, cv_name, filter_name, errata_types=None, date_type=None,
+            start_date=None, end_date=None, open_filter=True):
+        """Edit Erratum Date Range Filter"""
+        allowed_errata_types = FILTER_ERRATA_TYPE.values()
+        allowed_date_types = FILTER_ERRATA_DATE.values()
+        erratum_type_locator = locators['contentviews.erratum_type_checkbox']
+        date_type_locator = locators['contentviews.erratum_date_type']
+        if open_filter:
+            self.go_to_filter_page(cv_name, filter_name)
+        if errata_types is not None:
+            if errata_types:
+                # because of the behaviour of the UI to check and disable
+                # the last element if the others are unchecked,
+                # will check all the elements first,
+                # after then uncheck the not selected ones
+                for errata_type in allowed_errata_types:
+                    self.assign_value(erratum_type_locator % errata_type, True)
+                for errata_type in allowed_errata_types:
+                    if errata_type not in errata_types:
+                        self.assign_value(
+                            erratum_type_locator % errata_type, False)
+            else:
```

Comment:
OK. So i've passed `errata_types=['foo']`. This way, it's not `None`, it's not empty (i.e. `True` for second check), but nothing will be actually checked off and exception won't be risen. Think of different check instead of just `if errata_types:`